torch/csrc/autograd/engine.cpp

```diff
@@ -333,12 +330,11 @@ auto Engine::compute_dependencies(Function* root, GraphTask& task) -> void {
   while (queue.size() > 0) {
     auto fn = queue.back(); queue.pop_back();
     for (auto& edge : fn->next_functions) {
-      Function* next_ptr = edge.first.get();
-      if (!next_ptr) continue;
-      dependencies[next_ptr] += 1;
-      bool inserted;
-      std::tie(std::ignore, inserted) = seen.insert(next_ptr);
-      if (inserted) queue.push_back(next_ptr);
+      if (auto next_ptr = edge.function.get()) {
+        dependencies[next_ptr] += 1;
+        const auto result = seen.insert(next_ptr);
+        if (result.second) queue.push_back(next_ptr);
```

Comment:
nit: this has the downside that someone needs to remember what's `result.second`. It used to be clear, albeit more verbose before. The pattern I quite like now (which still makes the high-level idea clear) is
```cpp
if (/* bool unseen = */ seen.insert(next_ptr).second) queue.push_back(next_ptr);
```